tests/unit/dialog/core.js

```diff
@@ -84,6 +84,43 @@ QUnit.test( "ARIA", function( assert ) {
 	element.remove();
 } );
 
+QUnit.test( "aria-modal", function( assert ) {
+	assert.expect( 12 );
+
+	var element = $( "<div>" ).dialog( { modal: true } ),
+		wrapper = element.dialog( "widget" );
+	assert.equal( wrapper.attr( "aria-modal" ), "true", "modal option set to true, aria-modal attribute added" );
+	element.dialog( "option", "modal", false );
+	assert.equal( wrapper.attr( "aria-modal" ), undefined, "modal option set to false, aria-modal attribute not added" );
+	element.dialog( "option", "modal", null );
+	assert.equal( wrapper.attr( "aria-modal" ), undefined, "modal option not set, aria-modal attribute not added" );
```

Comment:
I guess we shouldn't really test setting `modal` to `null` since this is an option, not an `attr` call where that has a special meaning...
```suggestion
```